### PR TITLE
Add publisher

### PR DESCRIPTION
**Pull request by peteglondon**
Tests are passing.

App is running and you can create a thing.

Screenshot 2019-07-04 at 14 06 50

Haven't added e2e stack as I don't know what to put in it.

https://trello.com/c/gn3pWjbk/28-set-up-publisher

### CODE REVIEW

File: services/publisher/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      MONGODB_URI: "mongodb://mongo/publisher"
+      TEST_MONGODB_URI: "mongodb://mongo/publisher-test"
+      REDIS_URL: redis://redis
+      VIRTUAL_HOST: publisher.dev.gov.uk
+      HOST: 0.0.0.0
+    ports:
+      - "3000"
+    command: bundle exec unicorn -c ./config/unicorn.rb -p ${PORT:-3000}
+
+  publisher-worker:
```
Comment (benthorner) [inline services/publisher/docker-compose.yml]: Looks like nothing is using this - do we need it?

File: services/publisher/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      - router-app
+    environment:
+      MONGODB_URI: "mongodb://mongo/publisher"
+      TEST_MONGODB_URI: "mongodb://mongo/publisher-test"
+      REDIS_URL: redis://redis
+      VIRTUAL_HOST: publisher.dev.gov.uk
+      HOST: 0.0.0.0
+    ports:
+      - "3000"
+    command: bundle exec unicorn -c ./config/unicorn.rb -p ${PORT:-3000}
```
Comment (benthorner) [inline services/publisher/docker-compose.yml]: Would it be possible to use `rails s` here, like we do everywhere else?

File: services/publisher/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      - redis
+      - mongo
+      - calendars-app
+      - publishing-api-app
+      - asset-manager-app
+      - content-store-app
+      - router-app
+    environment:
+      MONGODB_URI: "mongodb://mongo/publisher"
+      TEST_MONGODB_URI: "mongodb://mongo/publisher-test"
```
Comment (benthorner) [inline services/publisher/docker-compose.yml]: We don't need test vars for app stacks.

File: services/publisher/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  rm ./google-chrome*.deb
+
+RUN wget https://bitbucket.org/ariya/phantomjs/downloads/phantomjs-2.1.1-linux-x86_64.tar.bz2
+RUN tar -xf phantomjs-2.1.1-linux-x86_64.tar.bz2
+RUN cp phantomjs-2.1.1-linux-x86_64/bin/phantomjs /usr/local/bin/phantomjs
+
+RUN curl -sL https://deb.nodesource.com/setup_12.x | bash -
+RUN apt-get update -qq && apt-get install -y nodejs
+
+RUN apt-get clean
```
Comment (benthorner) [inline services/publisher/Dockerfile]: We figured out this happens automatically, so we shouldn't need to do this :-).

**Comment by kevindew** [comment]
@benthorner I've picked this up and made quite a lot of changes, all your comments should now be covered.

Example of a published page: 
Screen Shot 2019-07-05 at 12 08 50

A few things:

- I noticed static was erroring due to lack of redis so I've resolved that.
- Calendar needs to publish routes before being usable so I've added that to setup.
- It didn't seem like link-checker-api had been working properly given it's set-up. I've been able to successfully check links.
- The publisher repo seems to have a lot of excess bits that don't seem to be used any more. As far as I can tell there is actually no need for Asset Manager and Imminence dependencies as they use models that are no longer used, though this is rather hard to confirm.

File: services/calendars/Makefile [hunk 5]
```diff
@@ -1,2 +1,3 @@
 calendars: $(GOVUK_ROOT_DIR)/calendars
 	$(COMPOSE_RUN) calendars-lite bundle
```
Comment (benthorner) [inline services/calendars/Makefile]: I'm doing a PR to add an explicit 'build' step for the 'lite' service. Can you add one in here?

Comment (benthorner) [reply services/calendars/Makefile]: https://github.com/alphagov/govuk-docker/pull/110

Comment (kevindew) [reply services/calendars/Makefile]: Done

**Pull request merged**